src/content/docs/en/guides/incremental-static-regeneration.mdx

```diff
@@ -0,0 +1,272 @@
+
+## Why not ISR?
+- Your content is located inside your project (for example, as markdown files in `src/content`).
+- You don't want the hassle of running a server or using a serverless platform.
+- Your website offers logged-in experiences. If each user needs to be sent a slightly different HTML from others, caching the page may not be possible.
+- The structure of your site (layouts, and styling) changes frequently. Changes to the structure require a full rebuild.
+
+## Implementing ISR for Your Project
+
+ISR involves saving the response such that it can be reused for future requests. The exact implementation depends on where you are deploying.
+
+If you aren't using an adapter already, you are going to need to add one. See the [integrations guide](/en/guides/integrations-guide/) for specific instructions. In your config, make sure that the `output` is either `"server"` or `"hybrid"`. Note that when using "hybrid", the pages you want generated, cached, and invalidated will need to be opted-out of prerendering. Otherwise, they will only be rendered once during build. See the [server side rendering guide](/en/guides/server-side-rendering/#opting-out-of-pre-rendering) to see how.
+
+#### About Cache-Control Headers
+You will see `Cache-Control` headers being used for many of the platforms below. It is worth reviewing the basics of how they work.
+ - `Cache-Control: s-maxage=60` tells the caching server to cache the response for 60 seconds. You probably want to use this for most cases.
+ - `Cache-Control: max-age=60` tells both browsers and servers to cache the response for 60 seconds. You may use this when your content is unlikely to be edited. If a page with `max-age` updates after it has been visited by a user, their browser will show them the cached version until the age of the content is past `max-age`.
+ - `Cache-Control: private, max-age=60` tells the browser to cache the response for 60 seconds, but tells the servers to not cache it. This is useful when you want to provide quick navigation, and the content includes details specific to the visitor.
+
+#### About the Web Cache API
+Deno and Cloudflare implement the Web Cache API, a way to store and retrieve `Response` objects using javascript.
+To use this API, call the `open` method on the globally available `caches` object with a name of your choosing.
+```ts
+const cache = await caches.open("astro")
+```
+You just opened a [`Cache`](https://developer.mozilla.org/en-US/docs/Web/API/Cache)! You can now call `await cache.put(request, response)` to save a response, and `await cache.match(request)` to retrieve it. See [web.dev's Cache API quick guide](https://web.dev/cache-api-quick-guide/) if you want to learn more.
+
+### Self-hosted Node.js
+ISR can be implemented in Node.js by writing a caching middleware. The following middleware caches every response in memory for 60 seconds:
+
+```ts title="src/middleware.ts"
+import { defineMiddleware } from "astro:middleware"

```

Comment:
I'm not sure if it is for a new API, but currently it gives me `package not found` with this identifier.

```suggestion
import { defineMiddleware } from "astro/middleware"

```

Reply:
I noticed there are more examples with this format `astro:middleware` in this PR

Reply:
This is how you import that function https://docs.astro.build/en/guides/middleware/.

Reply:
> This is how you import that function https://docs.astro.build/en/guides/middleware/.

https://stackblitz.com/edit/github-oiqdwp-dddxlh?file=src%2Fmiddleware.js,src%2Fpages%2Findex.astro

Although this is `"astro": "^2.10.14",` and here what it gives me:

```
> @example/basics@0.0.1 dev
> astro dev

  🚀  astro  v2.10.14 started in 94ms
  
  ┃ Local    http://localhost:3000/
  ┃ Network  use --host to expose
  
Error: The following dependencies are imported but could not be resolved:

  astro:middleware (imported by /home/projects/github-oiqdwp-dddxlh/src/middleware.js)

Are they installed?
    at eval (/home/projects/github-oiqdwp-dddxlh/node_modules/vite/dist/node/chunks/dep-df561101.js:45757:23)
    at async eval (/home/projects/github-oiqdwp-dddxlh/node_modules/vite/dist/node/chunks/dep-df561101.js:45166:38)
5:37:55 PM [vite] Error when evaluating SSR module /home/projects/github-oiqdwp-dddxlh/src//middleware: failed to import "astro:middleware"
|- Error: Cannot find module 'astro:middleware' imported from '/home/projects/github-oiqdwp-dddxlh/src/middleware.js'
    at nodeImport (/home/projects/github-oiqdwp-dddxlh/node_modules/vite/dist/node/chunks/dep-df561101.js:56062:25)
    at ssrImport (/home/projects/github-oiqdwp-dddxlh/node_modules/vite/dist/node/chunks/dep-df561101.js:55964:30)
    at eval (/home/projects/github-oiqdwp-dddxlh/src/middleware.js:3:50)
    at instantiateModule (/home/projects/github-oiqdwp-dddxlh/node_modules/vite/dist/node/chunks/dep-df561101.js:56026:15)

Failed to load url astro:middleware (resolved id: astro:middleware) in /home/projects/github-oiqdwp-dddxlh/src/middleware.js. Does the file exist?
```

Reply:
It is weird, cause locally seems to work just fine, you can even find the module declaration in the latest version of Astro:

```ts
declare module 'astro:middleware' {
	export * from 'astro/middleware/namespace';
}
```

Reply:
We introduced this alias in 3.0 for consistency with astro:assets and astro:content.

https://github.com/withastro/astro/blob/astro@3.0.0/packages/astro/CHANGELOG.md#minor-changes